File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.js
function solution(a, b, c) {
    let count = 1, answer = 1;
    
    if(a == b && b == c) count++;
    if(a == b || b == c || a == c) count++;
    for(let i = 1; i <= count; i++){
        answer *= Math.pow(a, i) + Math.pow(b, i) + Math.pow(c, i);
    }
    return answer;
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
class Solution {
    private int calculation(int a, int b, int c, int r){
        int result = 1;
        for(int i = 1; i <= r; i++){
            result *= Math.pow(a, i) + Math.pow(b, i) + Math.pow(c, i);
        }
        return result;
    }
    public int solution(int a, int b, int c) {
        if(a == b && a == c && b == c){
            return calculation(a, b, c, 3);
        }
        else if(a != b && a != c && b != c){
            return calculation(a, b, c, 1);
        }
        else return calculation(a, b, c, 2);
    }
}